src/rgw/store/dbstore/sqlite/sqliteDB.h

```diff
@@ -5,31 +5,49 @@
 #define SQLITE_DB_H
 
 #include <errno.h>
+#include <memory>
 #include <stdlib.h>
 #include <string>
 #include <sqlite3.h>
 #include "common/dbstore.h"
 
 using namespace rgw::store;
 
+using statement_ptr = std::unique_ptr<sqlite3_stmt, int(*)(sqlite3_stmt*)>;
+
 class SQLiteDB : public DB, virtual public DBOp {
   private:
     sqlite3_mutex *mutex = NULL;
 
+    // prepared statements
+    statement_ptr insert_user_stmt;
+
   protected:
     CephContext *cct;
 
   public:
     sqlite3_stmt *stmt = NULL;
     DBOpPrepareParams PrepareParams;
 
-    SQLiteDB(sqlite3 *dbi, std::string db_name, CephContext *_cct) : DB(db_name, _cct), cct(_cct) {
+    SQLiteDB(sqlite3 *dbi, std::string db_name, CephContext *_cct)
+        : DB(db_name, _cct),
+          insert_user_stmt(nullptr, &sqlite3_finalize),
```

Comment:
Could this be an initializer on the declaration above?  It seems to be constant.